sdk/storage/Azure.Storage.Blobs/src/BlobContainerClient.cs

```diff
@@ -1133,7 +1133,8 @@ private void SetNameFieldsIfNull()
                     return Response.FromValue(true, response);
                 }
                 catch (RequestFailedException storageRequestFailedException)
-                when (storageRequestFailedException.ErrorCode == BlobErrorCode.ContainerNotFound)
+                when (storageRequestFailedException.ErrorCode == BlobErrorCode.ContainerNotFound
+                || storageRequestFailedException.ErrorCode == BlobErrorCode.BlobNotFound)
```

Comment:
Should we be adding this check on other operations like `CreateIfNotExists` as well?

Reply:
I double checked, this is not an issue on ```FileSystemClient.CreateIfNotExists()```.